src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -170,26 +182,55 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
             }
         }
 
-        internal /*for testing purposes*/ void SilentSaveSolutionIfDirty()
+        internal /*for testing purposes*/ void DownloadBindingParameters(IProgressController controller, CancellationToken token, IProgressStepExecutionEvents notifications)
         {
-            bool saved = VsShellUtils.SaveSolution(this.host, silent: true);
-            Debug.Assert(saved, "Should not be cancellable");
+            notifications.ProgressChanged(Strings.PreparingBindingWorkflowProgessMessage, double.NaN);
+
+            var properties = this.host.SonarQubeService.GetProperties(token);
+
+            if (token.IsCancellationRequested)
+            {
+                AbortWorkflow(controller, token);
+                return;
+            }
+
+            var testProjRegexProperty = properties.FirstOrDefault(x => StringComparer.Ordinal.Equals(x.Key, ServerProperty.TestProjectRegexKey));
+            var testProjRegexPattern = testProjRegexProperty?.Value ?? ServerProperty.TestProjectRegexDefaultValue;
+
+            this.projectFilter.SetTestRegex(new Regex(testProjRegexPattern, RegexOptions.IgnoreCase));
```

Comment:
It will, yes. There's also no way to check if the expression is valid without using the ctor (and getting an Exception). I shall wrap in an `try-catch` and on failure proceed with...
1. the default, or
2. empty?